edb/schema/delta.py

```diff
@@ -3540,11 +3598,27 @@ def regular_cmd_from_ast(
                     schema=schema,
                 )
 
-            elif (isinstance(astnode.value, qlast.Set)
-                    and not astnode.value.elements):
+            elif (
+                isinstance(astnode.value, qlast.Set)
+                and not astnode.value.elements
+            ):
                 # empty set
                 new_value = None
 
+            elif isinstance(astnode.value, qlast.TypeExpr):
+                if not isinstance(parent_op, QualifiedObjectCommand):
+                    raise AssertionError(
+                        'cannot determine module for derived compound type'
+                        ' parent operation is not a QualifiedObjectCommand'
```

Comment:
I'm not really sure how to parse this sentence; maybe some punctuation is intended after the first line?
(Not that it really matters, as an assertion)

Reply:
Yeah, I missed a colon there